include/nccl_ofi_rdma.h

```diff
@@ -88,22 +113,24 @@ typedef struct nccl_net_ofi_rdma_mr_handle {
    destination buffer */
 typedef struct nccl_net_ofi_rdma_ctrl_msg {
 	/* Message type, must be NCCL_OFI_RDMA_MSG_CTRL */
-	nccl_ofi_rdma_msg_type_t type;
+	uint64_t type:NCCL_OFI_RDMA_CTRL_TYPE_BITS;
 
 	/* Message sequence number */
-	uint16_t msg_seq_num;
+	uint64_t msg_seq_num:NCCL_OFI_RDMA_SEQ_BITS;
 
 	/* A comm identitifer that uniquely identifies the comm
 	 * on the receiver side */
-	uint32_t remote_comm_id;
+	uint64_t remote_comm_id:NCCL_OFI_RDMA_COMM_ID_BITS;
+
+	uint64_t buff_len:NCCL_OFI_RDMA_BUFFLEN_BITS;
```

Comment:
I would like to insist that if we decide to go forward with bitfields, that we handle the little endian versus big endian case with ifdefs -- not because I think portability is super important here, but because I think it serves in and of itself as a useful comment that programmers need to think carefully about how they access these fields. Thoughts?

Reply:
In general, if you're looking for an alternative to bitfields, I think the other reasonable alternative is a flat array of uints, with a header of getters and setters. That would be my choice (I think bitfields are a feature better left unused), but this is bikeshedding and certainly the advantages of shrinking this outweigh any other concern.

Reply:
There's no reason to have little vs. big endian cases handled in an entire software stack that doesn't properly handle endianness.  Unless there's some correctness issue, you're just over-complicating a structure that should never be used for local storage, but only for something shoved onto the wire.

Reply:
+1 @bwbarrett  .       this entire stack is only been tested on little endian (x86-64 and aarch64) with 64B cacheline.    

big endian is pretty much something in the museums and text books and modern ML and HPC developers dont need to worry about it